test: Update the new ZTEST APIs tests/net/socket/poll

hunterhu3000 opened the pull request:
Move net/socket/poll tests to use new ztest API
TEST=twister -T tests/net/socket/poll

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

nashif commented on tests/net/socket/poll/src/main.c:
```diff
@@ -27,7 +27,7 @@ LOG_MODULE_REGISTER(net_test, CONFIG_NET_SOCKETS_LOG_LEVEL);
 #define FUZZ 10
 
-void test_poll(void)
+ZTEST(socket_poll_test_suite, test_poll)
```
no need for _test_suite here.

hunterhu3000 replied:
Updated. Thanks to review.

hunterhu3000 committed:
tests: net: socket: poll move to new ztest API

Move net/socket/poll tests to use new ztest API
TEST=twister -T tests/net/socket/poll

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

hunterhu3000 commented:
This test case cannot be run with CONFIG_ZTEST_SHUFFLE =y, as some of the test case cannot be run twice.
